lib/galaxy/workflow/extract.py

```diff
@@ -345,7 +345,7 @@ def cleanup( prefix, inputs, values ):
                 # Scrub dynamic resource related parameters from workflows,
                 # they cause problems and the workflow probably should include
                 # their state in workflow encoding.
-                if input.is_job_resource_conditional:
+                if input.name == '__job_resource':
                     if input.name in values:
```

Comment:
Ok, so I moved this customization out of the common conditional element code since it is only used for job resources and does only consists of the above line. I think it is easier to read too (: but that is probably subjective, so I am open for suggestions. Thanks a lot for your help. I really appreciate it.